contracts/core/Dispatcher.sol

```diff
@@ -250,7 +323,18 @@ contract Dispatcher is OwnableUpgradeable, UUPSUpgradeable, IDispatcher {
         bool feeEnabled,
         CounterParty calldata counterparty,
         Ics23Proof calldata proof
-    ) external {
+    ) external nonReentrant {
+        if (
+            bytes(local.portId).length == 0 || bytes(counterparty.portId).length == 0 || local.channelId == bytes32(0)
+                || counterparty.channelId == bytes32(0)
+        ) {
+            revert IBCErrors.invalidCounterParty();
+        }
+
+        if (address(receiver) == address(0)) {
+            revert IBCErrors.invalidAddress();
+        }
```

Comment:
Maybe move this to a function since it's also done in open-ack?

Reply:
ah yes good one; should be done now